rspace/src/main/scala/coop/rchain/rspace/ReportingRspace.scala

```diff
@@ -105,31 +125,23 @@ class ReportingRspace[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C, P,
   implicit protected[this] override lazy val MetricsSource: Metrics.Source =
     Metrics.Source(RSpaceMetricsSource, "reporting")
 
-  /**
-    * in order to distinguish the system deploy(precharge and refund) in the a normal user deploy
-    * It might be more easily to analyse the report with data structure
-    * Seq(Seq[ReportingEvent](Precharge), Seq[ReportingEvent](userDeploy), Seq[ReportingEvent](Refund))
-    * It would be seperated by the softcheckpoint creation.
-    */
-  val report: SyncVar[Seq[Seq[ReportingEvent]]] = create[Seq[Seq[ReportingEvent]]](Seq.empty)
-  val softReport: SyncVar[Seq[ReportingEvent]]  = create[Seq[ReportingEvent]](Seq.empty)
-
   private def collectReport =
     for {
       sReport <- getSoftReport
-      _ = if (sReport.nonEmpty) {
-        report.update(s => s :+ sReport)
-        softReport.update(_ => Seq.empty[ReportingEvent])
-      }
+      _ <- if (sReport.nonEmpty) for {
+            _ <- report.update(s => s :+ sReport)
+            _ <- softReport.update(_ => Seq.empty[ReportingEvent])
+          } yield ()
+          else Unit.pure[F]
     } yield ()
 
   def getReport: F[Seq[Seq[ReportingEvent]]] =
     for {
       _      <- collectReport
-      result = report.get
+      result <- report.get
       _      = report.update(_ => Seq.empty[Seq[ReportingEvent]])
```

Comment:
This is not executing any effect

Reply:
I didn't understand this comment. Here I am reading the contents of the report before erasing to return it as a result of for-comprehension.
As variant, I can replace whole this for-comprehension with
`collectReport.flatMap(_ => report.modify(r => (Seq.empty[Seq[ReportingEvent]], r)))`
but I realy don't undestand the difference.